libraries/core/src/bin/generate_schema.rs

```diff
@@ -0,0 +1,38 @@
+use std::{env, path::Path};
+
+use dora_core::descriptor::Descriptor;
+use schemars::schema_for;
+
+fn main() -> () {
+    let schema = schema_for!(Descriptor);
+    let raw_schema =
+        serde_json::to_string_pretty(&schema).expect("Could not serialize schema to json");
+    let raw_schema = raw_schema.replace(
+        "\"additionalProperties\": false",
+        "\"additionalProperties\": true",
+    );
```

Comment:
Why is this replace needed?

Reply:
// Add additional properties to True, as #[derive(transparent)] of enums are not well handled.
    //
    // 'OneOf' such as Custom Nodes, Operators and Single Operators overwrite property values of the initial struct `Nodes`.`
    // which make the original properties such as `id` and `name` not validated by IDE extensions.
    
I have added documentation but basically its because we use #[derive(transparent)] on an enum, and the parent properties get forgotten in the JSON Schema.